### PR TITLE
Fix #25926 Reimplement the autocomplete component

### LINKED ISSUE #25926: Templater Builder: Fixing Wrong Classes Deletion
### Problem Statement
In templater builder when remove classes from box and rows it remove the wrong one.

### Steps to Reproduce
1. Edit a template
2. Add classes to box or row
3. Edit the classes and remove first one
4. See the error how delete the last one.

### Acceptance Criteria
1. User neends to be able to delete the clicked chip
2. System should remove the correct class
3. Remove the chip from the box or rows

### External Links
N/A

### Assumptions & Initiation Needs
N/A

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by fmontes**
### Summary
:recycle::sparkles::test_tube:

Refactored the `add-style-classes-dialog` component to use a JSON file and a p-autoComplete element to display and select the style classes. Removed the store logic and some redundant styles and attributes from the component. Added a new service and a spec file for fetching the style classes from the JSON file.

> _We're breaking free from the store of doom_
> _We're fetching data from the JSON tomb_
> _We're simplifying the `p-autoComplete`_
> _We're adding style to the template elite_

### Walkthrough
*  Simplified the HTML template of the add-style-classes-dialog component to make it more user-friendly and consistent with the design mockups ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-ad38fd507e948aab0702b09038a51398d2751e9070676c7c562182f78748eacdL1-R21))
*  Removed the store logic and the unnecessary lifecycle hooks from the TS file of the add-style-classes-dialog component and used the JsonClassesService instead to fetch the style classes from a JSON file ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-5bf16051f3849387c22bfabb3e46611bbc31a29a300e848e3af205b38d416674L1-R12), [link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-5bf16051f3849387c22bfabb3e46611bbc31a29a300e848e3af205b38d416674L30-R28), [link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-5bf16051f3849387c22bfabb3e46611bbc31a29a300e848e3af205b38d416674L49-R53))
*  Added the JsonClassesService and its spec files to implement the service logic and the unit testing for the component ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-5643fe4db24fda08277deb5b0eb768fe042aecebf3b615910646b5a67fe696c8R1-R16), [link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-acc46fa2a4b22eafe3eb9243e90f0cc5480dd4eed33bf341de9a51e8ec09b48fR1-R15))
*  Cleaned up the SCSS file of the add-style-classes-dialog component by removing the unused styles ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-8b84f5fdb2071a26374ae41f88ef831a4ddb6370be8990650031ca7cc0ed1b37L7-R11))
*  Removed the DotAddStyleClassesDialogStore import and provider from the template-builder module ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-f5608bc0e0cfd9dc606805e064c3a3de96ed347a844c8ef22f9ef831a0b80522L11), [link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-f5608bc0e0cfd9dc606805e064c3a3de96ed347a844c8ef22f9ef831a0b80522L30-R29))
*  Deleted the DotAddStyleClassesDialogStore and its spec files as they are no longer needed ([link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-3751e6d44fc59b72586e1e24c65228bfe0bc265f66e88807f5e66f10b8b69a83), [link](https://github.com/dotCMS/core/pull/25935/files?diff=unified&w=0#diff-f08eb5ab3c6c8b4254e1b0e6468e42c2341efe751de6fd6d1b27a1ac910443e4))

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

https://github.com/dotCMS/core/assets/751424/0014f560-de1a-4485-8ec8-ea8910ba7072

### CODE REVIEW

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/mocks.ts [hunk 1]
```diff
@@ -288,7 +288,9 @@ export const MESSAGES_MOCK = {
     'editpage.layout.theme.search': 'Search',
     'dot.template.builder.classes.dialog.update.button': 'Update',
     'dot.template.builder.sidebar.header.title': 'Sidebar',
-    'dot.template.builder.row.box.wont.fit': 'Minimum 1 column needed for box drop.'
+    'dot.template.builder.row.box.wont.fit': 'Minimum 1 column needed for box drop.',
+    'dot.template.builder.autocomplete.has.suggestions': 'Add the message for suggestions case',
+    'dot.template.builder.autocomplete.no.suggestions': 'Add the message for no suggestions case'
```
Comment (zJaaal) [inline core-web/libs/template-builder/src/lib/components/template-builder/utils/mocks.ts:293]: Update this messages for the story and also add the missing ones

Comment (zJaaal) [reply core-web/libs/template-builder/src/lib/components/template-builder/utils/mocks.ts:293]: You can use this object in the test for the messages.

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.scss [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
-        .p-autocomplete-multiple-container
-        .p-autocomplete-token {
-        margin-right: 0;
-    }
+p-autoComplete {
+    margin-bottom: $spacing-3;
+    display: block;
+}
+
+ul {
```
Comment (manuelrojas) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.scss:18]: is it safe to apply directly to all ul?

File: core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.scss [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
+    display: flex;
+    flex-direction: column;
+    font-size: $font-size-sm;
+    gap: $spacing-1;
+    list-style: none;
+    margin: 0;
+    padding: 0;
+}
+
+li {
```
Comment (manuelrojas) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/add-style-classes-dialog/add-style-classes-dialog.component.scss:29]: I wonder the same here, is it safe to apply directly to all li.

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/mocks.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/template-builder/src/lib/components/template-builder/utils/mocks.ts:293]: Not needed for tests... overkill

**Review by manuelrojas — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Pull request merged**